deepchem/models/torch_models/lcnn.py

```diff
@@ -0,0 +1,428 @@
+
+
+class LCNN(nn.Module):
+  """
+  The Lattice Convolution Neural Network (LCNN)
+
+  This model takes lattice representation of Adsorbate Surface to predict
+  coverage effects taking into consideration the adjacent elements interaction
+  energies.
+
+  The model follows the following steps
+
+  [1] It performs n lattice convolution operations. For more details look at the LCNNBlock class
+  [2] Followed by Linear layer transforming into sitewise_n_feature
+  [3] Transformation to scalar value for each node.
+  [4] Average of properties per each element in a configuration
+
+  Refrences
+  -----------
+
+  [1] Jonathan Lym,Geun Ho Gu, Yousung Jung , and Dionisios G. Vlachos
+  "Lattice Convolutional Neural Network Modeling of Adsorbate Coverage
+  Effects" The Journal of Physical Chemistry
+
+  [2] https://forum.deepchem.io/t/lattice-convolutional-neural-network-modeling-of-adsorbate-coverage-effects/124
+
+  Notes
+  -----
+  This class requires DGL and PyTorch to be installed.
+  """
+
+  def __init__(self,
+               n_occupancy: int = 3,
+               n_neighbor_sites: int = 19,
+               n_permutation: int = 6,
+               dropout_rate: float = 0.2,
+               n_conv: int = 2,
+               n_features: int = 19,
+               sitewise_n_feature: int = 25):
+    """
+    parameters
+    ----------
+
+    n_occupancy: int
+        number of possible occupancy
+    n_neighbor_sites_list: int
+        Number of neighbors of each site.
+    n_permutation: int
+        Diffrent permutations taken along diffrent directions.
+    dropout_rate: float
+        p value for dropout between 0.0 to 1.0
+    nconv: int
+        number of convolutions performed
+    n_feature: int
+        number of feature for each site
+    sitewise_n_feature: int
+        number of features for atoms for site-wise activation
+
+    """
+    super(LCNN, self).__init__()
+
+    modules = [LCNNBlock(n_occupancy * n_neighbor_sites, n_features)]
+    for i in range(n_conv - 1):
+      modules.append(
+          LCNNBlock(n_features * n_neighbor_sites, n_features, n_permutation))
+
+    self.LCNN_blocks = nn.Sequential(*modules)
+    self.Atom_wise_Conv = Atom_Wise_Convolution(n_features, sitewise_n_feature)
+    self.Atom_wise_Lin = AtomWiseLinear(sitewise_n_feature, sitewise_n_feature)
+
+  def forward(self, G):
+    """
+    Parameters
+    ----------
+    G: DGLGraph
+        DGLGraph for a batch of graphs.
+
+    Returns
+    -------
+    y: torch.Tensor
+        A single scalar value
+
+    """
+    try:
+      import dgl
+    except:
+      raise ImportError("This class requires DGL to be installed.")
+    node_feats = G.ndata.pop('x')
+
+    for conv in self.LCNN_blocks:
+      node_feats = conv(G, node_feats)
+    node_feats = self.Atom_wise_Conv(node_feats)
+    node_feats = self.Atom_wise_Lin(node_feats).sum(axis=1)
+    G.ndata['new'] = node_feats
+
+    y = dgl.mean_nodes(G, 'new')
+    return y
+
+
+class LCNNModel(TorchModel):
```

Comment:
Can you add unit tests and ML correctness tests for this class? See

https://deepchem.readthedocs.io/en/latest/development_guide/coding.html#unit-tests
https://deepchem.readthedocs.io/en/latest/development_guide/coding.html#testing-machine-learning-models

for a guide on the tests that we'll need. In particular, we'll need an overfitting test and a reloading test for `LCNNModel`